nativelib/src/main/scala/scala/scalanative/runtime/DeleteOnExit.scala

```diff
@@ -0,0 +1,22 @@
+package scala.scalanative.runtime
+
+import scala.collection.mutable
+import scala.scalanative.native.{Zone, stdio, toCString}
+
+object DeleteOnExit {
+  private[this] val toDeleteSet: mutable.Set[String] = mutable.Set.empty
+  private[this] val toDelete: mutable.ArrayBuffer[String] =
+    mutable.ArrayBuffer.empty
+  Shutdown.addHook(2, new Runnable() {
+    override def run() {
+      toDelete.foreach { f =>
+        Zone { implicit z =>
+          stdio.remove(toCString(f))
+        }
+      }
+    }
+  })
+  def add(s: String) = toDelete.synchronized {
```

Comment:
It would be more clear with more explicit names here `def deleteOnExit(path: String): Unit = ...`

Reply:
`DeleteOnExit.deleteOnExit` sounds redundant to me. Perhaps `DeleteOnExit.addFile(name: String)`?

Reply:
That works as well.